common/beerocks/scripts/prplmesh_utils.sh.in

```diff
@@ -315,7 +316,22 @@ main() {
             ;;
         "start_wait")
             start_function
-            wait -n
+
+            # Store PIDs of the Controller and Agents
+            PREV_PIDS=$(pgrep "beerocks_(agent|controller)")
+            CURR_PIDS=$PREV_PIDS
+            
+            # Monitor the number of running processes
+            while [ ${#CURR_PIDS[@]} -ge ${#PREV_PIDS[@]} ]; do
```

Comment:
`${#foo[@]}` is a bashism, and we use /bin/sh. Not OK...

You can just sort the output of pgrep and compare the variables as strings. Or you can wc -l it and compare the integers.

Reply:
Bashism or not, if it works in the ash shell as well, why not keep it? :)

Reply:
I agree with @arnout.
Since this particular script can potentially be used on many different platforms, you can't rely on the fact that some specific implementations do support it.

Reply:
Same assumption is true to many other parts of the this script.
Is it safe to assume that all platform support `pgrep` or `wc`?

Reply:
True, except that you can make it a platform requirement to have, say `procps` available.
You can also make it a requirement to have bash installed for instance (although we don't want to impose that), but it doesn't mean that `/bin/sh` will run bash.
Since `/bin/sh` could really be any shell, the general common denominator we can assume is that it's going to be a POSIX-compliant shell.

The problem is not about the availability of the tools being used, but rather about using features available across different implementations of the same tool (the shell in this case). In the cases where we can, why not do it?

Reply:
No reason, I was just curious about where we'd like to draw the compatibility line.
Pushed a fixup with all the changes.